core/services/run_manager.go

```diff
@@ -261,7 +259,8 @@ func (rm *runManager) Create(
 		return nil, fmt.Errorf("invariant for job %s: no tasks to run in NewRun", job.ID)
 	}
 
-	run, adapters := NewRun(&job, initiator, data, creationHeight, runRequest, rm.config, rm.orm, now)
+	runRequest.RequestParams = *data
```

Comment:
Can we avoid modifying the argument here?  runRequest is passed as a reference, and it could be considered unexpected behavior by the caller.

Reply:
Also what about returning an error if *data is nil?  It could prevent a potential panic.

Reply:
This just moved from somewhere else. The likelihood of panic is unchanged.

Reply:
And the modification of the runRequest argument?

Reply:
We cannot avoid modifying the argument. `data` is called from within the function.

Reply:
NOTE to self: run request is modified in the wrong place. Fix this.